### DISCUSSION TITLE
Avoid reload of Live Preview iframe on device type selection

### DISCUSSION THREAD

**Discussion by membla** [craftcms/cms#10353] [Ideas] [answered]
The iframe containing the live preview target is reloaded when changing device preview – is this always necessary? Having the option to not reload the iframe would be very welcome.

![image](https://user-images.githubusercontent.com/1955910/148960517-047b4298-fe4a-466e-8f20-f006c63a9fee.png)

**Comment by brandonkelly** [comment]
@joshangell any thoughts on this?

**Reply by joshangell** [reply]
I’ve just had a quick refresher of the code, and it seems this line is the one triggering it: https://github.com/craftcms/cms/blob/develop/src/web/assets/cp/src/js/Preview.js#L604

You could try omitting the second parameter and see if nothing breaks, but there is probably a reason it forces a reload - I just failed to comment it ...

**Comment by brandonkelly** [comment, marked as answer]
Got this working for the next release! (26d3c8e24191f1c01ebd9bf9e8bfb701d1cea39a)

**Reply by joshangell** [reply]
Well that was more complex than I thought!

**Reply by membla** [reply]
> Got this working for the next release! ([26d3c8e](https://github.com/craftcms/cms/commit/26d3c8e24191f1c01ebd9bf9e8bfb701d1cea39a))

Fantastic, thank you so much Brandon! 🏆

**Reply by brandonkelly** [reply]
Craft 3.7.29 is out now with that change ✨
